src/nvim/ex_getln.c

```diff
@@ -1447,6 +1447,9 @@ static int command_line_handle_key(CommandLineState *s)
              && ccline.cmdbuff[ccline.cmdpos - 1] != ' ');
 
     set_cmdspos_cursor();
+    if (ccline.special_char != NUL) {
+      putcmdline(ccline.special_char, ccline.special_shift);
+    }
```

Comment:
@LTKills  Why do this here, in the `K_LEFT`/`K_S_LEFT`/... case? I don't see anything like that in https://github.com/vim/vim/commit/a92522fbf3a49d06e08caf010f7d7b0f58d2e131 .